src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -0,0 +1,227 @@
+    pt.put("error_msg", ex.what());
+  }
+
+  return pt;
+}
+
+void
+ReportAie::getPropertyTreeInternal( const xrt_core::device * _pDevice, 
+                                              boost::property_tree::ptree &_pt) const
+{
+  // Defer to the 20202 format.  If we ever need to update JSON data, 
+  // Then update this method to do so.
+  getPropertyTree20202(_pDevice, _pt);
+}
+
+void 
+ReportAie::getPropertyTree20202( const xrt_core::device * _pDevice, 
+                                           boost::property_tree::ptree &_pt) const
+{
+  _pt.add_child("aie_metadata", populate_aie(_pDevice, "Aie_Metadata"));
+}
+
+void 
+ReportAie::writeReport( const xrt_core::device * _pDevice,
+                                  const std::vector<std::string> & /*_elementsFilter*/, 
+                                  std::iostream & _output) const
+{
+  boost::property_tree::ptree _pt;
+  boost::property_tree::ptree empty_ptree;
+  getPropertyTreeInternal(_pDevice, _pt);
+
+  _output << "Aie\n";
+  _output << boost::format("  %-10s\n") % _pt.get<std::string>("aie_metadata.description");
+
+  try {
+    for (auto& gr: _pt.get_child("aie_metadata.graphs")) {
+      boost::property_tree::ptree& graph = gr.second;
+      _output << boost::format("  GRAPH[%2d] Name:%2s\n") % graph.get<std::string>("id")
+           % graph.get<std::string>("name");
+      _output << boost::format("    SNo.  %-20s%-30s%-30s\n") % "Core [C:R]"
+           % "Iteration_Memory [C:R]" % "Iteration_Memory_Addresses";
+      int count = 0;
+      for (auto& node : graph.get_child("tile")) {
+        boost::property_tree::ptree& tile = node.second;
+        _output << boost::format("    [%2d]   %-20s%-30s%-30d\n") % count
+            % (tile.get<std::string>("col")+":"+tile.get<std::string>("row"))
+            % (tile.get<std::string>("memcol")+":"+tile.get<std::string>("memrow"))
+            % node.second.get<uint16_t>("memaddr");
+        count++;
+      }
+      _output << std::endl;
+
+      _output << boost::format("    %s\n") % "Pl Kernel Instances in Graph:";
+      for (auto& node : graph.get_child("pl_kernel")) {
+        _output << boost::format("      %s\n") % node.second.data();
+      }
+      _output << std::endl;
+    }
+
+    int count = 0;
+    for (auto& rtp_node : _pt.get_child("aie_metadata.rtps")) {
+      _output << boost::format("  %-3s:[%2d]\n") % "RTP" % count;
+      _output << boost::format("    %-22s: %s\n") % "Port Name" % rtp_node.second.get<std::string>("port_name");
+      _output << boost::format("    %-22s: %d\n") % "Selector Row" % rtp_node.second.get<uint16_t>("selector_row");
+      _output << boost::format("    %-22s: %d\n") % "Selector Col" % rtp_node.second.get<uint16_t>("selector_column");
+      _output << boost::format("    %-22s: %d\n") % "Selector Lock Id" % rtp_node.second.get<uint16_t>("selector_lock_id");
+      _output << boost::format("    %-22s: %d\n") % "Selector Addr" %  rtp_node.second.get<uint64_t>("selector_address");

```

Comment:
Normally addresses being reported in a hex format and not decimal.  Is there a reason why in this case it is decimal?